web/quiz_views.py

```
@@ -666,16 +947,16 @@
 
     # Calculate overall statistics
     total_attempts = attempts.count()
-    average_score = attempts.exclude(score=None).values_list("score", flat=True)
+    all_attempts = list(attempts)
+    average_score = [attempt.calculate_score() for attempt in all_attempts if attempt.calculate_score() is not None]
     if average_score:
         average_score = sum(average_score) / len(average_score)
     else:
         average_score = 0
```

Ruff (0.8.2)

Use ternary operator `average_score = sum(average_score) / len(average_score) if average_score else 0` instead of `if`-`else`-block

Replace `if`-`else`-block with `average_score = sum(average_score) / len(average_score) if average_score else 0`

(SIM108)